### PR TITLE
Added use of some discount components to sample app

### PR DESCRIPTION
**Pull request by nickwesselman**
* `MethodCard`, `ActiveDatesCard`, `CombinationCard`, `UsageLimitsCard`
* Also enables creation of both code and automatic discounts in the process

### CODE REVIEW

File: sample-apps/discount-functions-sample-app/web/frontend/components/DiscountDetailsPage.jsx [hunk 1]
```diff
@@ -74,45 +103,91 @@ export default function DiscountDetailsPage({
...
+            <Card>
+              <Card.Section>
+                {renderConfigurationForm(configuration, setConfiguration)}
+              </Card.Section>
+            </Card>
+            {method === DiscountMethod.Code && (
+              <UsageLimitsCard
+                totalUsageLimit={{
+                  value: usageLimit,
+                  onChange: (value) => setUsageLimit(parseInt(value)),
```
Comment (lhoffbeck) [inline sample-apps/discount-functions-sample-app/web/frontend/components/DiscountDetailsPage.jsx:136]: (we talked about this in slack but leaving it here too) This breaks the input, it expects a string. Suggest changing to `onChange: setUsageLimit` and casting to an int in the submit mutation

File: sample-apps/discount-functions-sample-app/web/frontend/components/DiscountCreatePage.jsx [hunk 2]
```diff
@@ -56,19 +83,62 @@ export default function DiscountCreatePage({
...
-            </Card.Section>
             <Card.Section>
               {renderConfigurationForm(configuration, setConfiguration)}
             </Card.Section>
           </Card>
+          {method === DiscountMethod.Code && (
+            <UsageLimitsCard
+              totalUsageLimit={{
+                value: usageLimit,
+                onChange: (value) => setUsageLimit(parseInt(value)),
```
Comment (lhoffbeck) [inline sample-apps/discount-functions-sample-app/web/frontend/components/DiscountCreatePage.jsx:111]: (we talked about this in slack but leaving it here too) This breaks the input, it expects a string. Suggest changing to `onChange: setUsageLimit` and casting to an int in the submit mutation

File: sample-apps/discount-functions-sample-app/web/frontend/components/providers/DiscountProvider.jsx [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { AppProvider } from '@shopify/discount-app-components'
+import '@shopify/discount-app-components/build/esm/styles.css'
+
+export function DiscountProvider({ children }) {
+  return <AppProvider locale="en-US" ianaTimezone="America/Toronto">{children}</AppProvider>
```
Comment (andrewhassan) [inline sample-apps/discount-functions-sample-app/web/frontend/components/providers/DiscountProvider.jsx:5]: Should these also be from the current user?

File: sample-apps/discount-functions-sample-app/web/frontend/components/DiscountDetailsPage.jsx [hunk 4]
```diff
@@ -74,45 +103,91 @@ export default function DiscountDetailsPage({
...
+            <ActiveDatesCard
+              startDate={{
+                value: startsAt,
+                onChange: setStartsAt,
+              }}
+              endDate={{
+                value: endsAt,
+                onChange: setEndsAt,
+              }}
+              timezoneAbbreviation="EST"
```
Comment (andrewhassan) [inline sample-apps/discount-functions-sample-app/web/frontend/components/DiscountDetailsPage.jsx:165]: Same question here

**Pull request closed without merging**